File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Ram/AlibabaCloud/SDK/ROS/CDK/Ram/IGrantable.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Ram
{
    /// <summary>Any object that has an associated principal that a permission can be granted to.</summary>
    [JsiiInterface(nativeType: typeof(IGrantable), fullyQualifiedName: "@alicloud/ros-cdk-ram.IGrantable")]
    public interface IGrantable
    {
        /// <summary>The principal to grant permissions to.</summary>
        [JsiiProperty(name: "grantPrincipal", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-ram.IPrincipal\"}")]
        AlibabaCloud.SDK.ROS.CDK.Ram.IPrincipal GrantPrincipal
        {
            get;
        }

        /// <summary>Any object that has an associated principal that a permission can be granted to.</summary>
        [JsiiTypeProxy(nativeType: typeof(IGrantable), fullyQualifiedName: "@alicloud/ros-cdk-ram.IGrantable")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Ram.IGrantable
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>The principal to grant permissions to.</summary>
            [JsiiProperty(name: "grantPrincipal", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-ram.IPrincipal\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Ram.IPrincipal GrantPrincipal
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Ram.IPrincipal>()!;
            }
        }
    }
}

File: multiple-languages/java/ros-cdk-ram/src/main/java/com/aliyun/ros/cdk/ram/IGrantable.java
package com.aliyun.ros.cdk.ram;

/**
 * Any object that has an associated principal that a permission can be granted to.
 */
@javax.annotation.Generated(value = "jsii-pacmak/1.85.0 (build 08ee592)", date = "2025-09-28T10:11:28.380Z")
@software.amazon.jsii.Jsii(module = com.aliyun.ros.cdk.ram.$Module.class, fqn = "@alicloud/ros-cdk-ram.IGrantable")
@software.amazon.jsii.Jsii.Proxy(IGrantable.Jsii$Proxy.class)
public interface IGrantable extends software.amazon.jsii.JsiiSerializable {

    /**
     * The principal to grant permissions to.
     */
    @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.ram.IPrincipal getGrantPrincipal();

    /**
     * A proxy class which represents a concrete javascript instance of this type.
     */
    @software.amazon.jsii.Internal
    final class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements com.aliyun.ros.cdk.ram.IGrantable.Jsii$Default {
        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
            super(objRef);
        }

        /**
         * The principal to grant permissions to.
         */
        @Override
        public final @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.ram.IPrincipal getGrantPrincipal() {
            return software.amazon.jsii.Kernel.get(this, "grantPrincipal", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.ram.IPrincipal.class));
        }
    }

    /**
     * Internal default implementation for {@link IGrantable}.
     */
    @software.amazon.jsii.Internal
    interface Jsii$Default extends IGrantable {

        /**
         * The principal to grant permissions to.
         */
        @Override
        default @org.jetbrains.annotations.NotNull com.aliyun.ros.cdk.ram.IPrincipal getGrantPrincipal() {
            return software.amazon.jsii.Kernel.get(this, "grantPrincipal", software.amazon.jsii.NativeType.forClass(com.aliyun.ros.cdk.ram.IPrincipal.class));
        }
    }
}
